### ISSUE TITLE
Use a queue with enforceFocusVicinity.

### ISSUE THREAD

**Issue by samussiah** [samussiah/force-directed-graph#235]
i.e. add states to a node's queue as the animation reaches the appropriate timepoint and drop the state from the queue once the node reaches the associated focus.  Using a queue will allow each node to visually travel to each focus.

Right now if a node has multiple states in close proximity and is still traveling to the first state it just kind of ignores the following states on its way to the first state and skips to the next state at the timepoint that it reaches the focus.

Basically the node forgets the states that fall after the focus it's traveling to and the state it should be in once it reaches that first focus.
